sql/tests: internal error: ordering column group contains non-equivalent columns

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/14090206?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/14090206?buildTab=artifacts#/) on release-23.2.1-rc @ [898cd6a363fd47bb92a03bac216f9bed0f64bc08](https://github.com/cockroachdb/cockroach/commits/898cd6a363fd47bb92a03bac216f9bed0f64bc08):

Random syntax error:

```
    rsg_test.go:911: Crash detected: server panic: pq: internal error: ordering column group (84,87,89) contains non-equivalent columns (op sort)
```
Query:

```
        SELECT
        	" tab�183065".tableoid AS "col\\xdc405653",
        	(
        		SELECT
        			NULL AS "c'ol405664"
        		FROM
        			defaultdb.public.seed@[0] AS tab183068
        			JOIN defaultdb.public.table1@[0] AS "ta'b183069"
        				JOIN (
        						SELECT
        							tab183064."co\\U0002A8F0l4_2" AS "co l405654",
        							(-9.85070379389495794E+30):::DECIMAL AS col405655,
        							true AS col405656,
        							ARRAY[0:::DECIMAL] AS col405657,
        							tab183063.crdb_internal_mvcc_timestamp AS co😑l405658
        						FROM
        							defaultdb.public.table2@[0] AS tab183070
        					)
        						AS tab183071 (c😚ol405659, col405660, "%vc😄ol405661", col405662, col405663) ON
        						("ta'b183069"."col1\\U000B9365_ 1") = (tab183071.c😚ol405659)
        						AND ("ta'b183069"."col1_�10") = (tab183071.c😚ol405659)
        						AND ("ta'b183069"."Col1 _
        6") = (tab183071.c😚ol405659)
        						AND ("ta'b183069".col1_7) = (tab183071.c😚ol405659)
        						AND ("ta'b183069".col1_4) = (tab183071.c😚ol405659) ON
        					(tab183068._string) = ("ta'b183069"."Col1 _
        6") AND (tab183068._string) = (tab183071.c😚ol405659)
        		ORDER BY
        			tab183068._float4 ASC NULLS FIRST,
        			tab183068._interval,
        			"ta'b183069"."Col1 _
        6" DESC NULLS LAST,
        			"ta'b183069".col1_3 DESC,
        			tab183068._inet DESC
        		LIMIT
        			1:::INT8
        	)
        		AS col405665,
        	(-28391):::INT8 AS "c%qo""😘l405666",
        	'71d60280-c314-44f5-adfd-1ee2b72a0d1e':::UUID AS "c\\xb5ol405667",
        	NULL AS col405668,
        	" tab�183065"._interval AS col405669,
        	tab183064."c'ol4_0" AS col405670,
        	'hZW':::STRING AS "cOl405671",
        	tab183063."co'l1_""5" AS col405672,
        	'':::STRING AS "c.ol405673",
        	" tab�183065".crdb_internal_mvcc_timestamp AS col405674,
        	tab183063.col1_3 AS "col%2e405675",
        	" tab183067".tableoid AS "co&l405676",
        	'2005-02-19 07:16:45.000819+00':::TIMESTAMPTZ AS c😝ol405677,
        	" tab�183065".tableoid AS col405678,
        	false AS "c.ol405679"
        FROM
        	defaultdb.public.table1@[0] AS tab183063,
        	defaultdb.public.table4@[0] AS tab183064,
        	defaultdb.public.seed@[0] AS " tab�183065"
        	JOIN defaultdb.public."ta̹""blE%v3"@[0] AS tab183066
        		JOIN defaultdb.public.table4@"table4_co%pl4_1_key" AS " tab183067" ON
        				(tab183066.tableoid) = (" tab183067".tableoid)
        				AND (tab183066.crdb_internal_mvcc_timestamp) = (" tab183067".crdb_internal_mvcc_timestamp) ON
        			(" tab�183065"._uuid) = (tab183066.c😾ol3_1)
        			AND (" tab�183065".crdb_internal_mvcc_timestamp) = (tab183066.crdb_internal_mvcc_timestamp)
        WHERE
        	true
        LIMIT
        	14:::INT8;
```
Schema:

```
    rsg_test.go:720: To reproduce, use schema:
    rsg_test.go:722: SET sql_safe_updates = false;;
    rsg_test.go:722: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:722: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:722: CREATE TABLE table1 (col1_0 DATE NOT NULL, "col1\\U000B9365_ 1" CHAR NOT NULL, "c\\U000CA96Fol1_2" INT4 NOT NULL, col1_3 INT4 NOT NULL, col1_4 NAME NOT NULL, "co'l1_""5" BIT(32) NOT NULL, "Col1 _
        6" STRING NOT NULL AS (lower("col1\\U000B9365_ 1")) STORED, col1_7 STRING NOT NULL AS (CASE WHEN col1_0 IS NULL THEN e'0\x1f\x7fw3\x14_e':::STRING ELSE '':::STRING END) VIRTUAL, co̿l1_8 INT8 NOT NULL AS ("c\\U000CA96Fol1_2" + col1_3) STORED, c̝ol1_9 INT8 NOT NULL AS (col1_3 + "c\\U000CA96Fol1_2") STORED, "col1_�10" STRING NOT NULL AS (lower(CAST("co'l1_""5" AS STRING))) VIRTUAL, UNIQUE (("c\\U000CA96Fol1_2" + col1_3) ASC, col1_7 DESC), INDEX (col1_4 DESC) STORING (col1_0, "c\\U000CA96Fol1_2", "co'l1_""5", "Col1 _
        6", co̿l1_8) WHERE (((((table1."col1_�10" <= '':::STRING) OR (table1.c̝ol1_9 < 0:::INT8)) OR (table1."c\\U000CA96Fol1_2" != (-1):::INT8)) AND (table1.col1_0 = '1970-01-01':::DATE)) OR (table1."Col1 _
        6" = '"':::STRING)) AND (table1."col1\\U000B9365_ 1" > 'X':::STRING), UNIQUE ("col1_�10", c̝ol1_9 DESC, col1_4, (col1_3 + "c\\U000CA96Fol1_2") DESC, "col1\\U000B9365_ 1" DESC, col1_7 DESC, co̿l1_8 DESC, col1_3, "co'l1_""5" DESC) WHERE (table1.co̿l1_8 >= 2147483647:::INT8) AND (table1.c̝ol1_9 >= 1:::INT8), UNIQUE ("Col1 _
        6" ASC) STORING (col1_0, "col1\\U000B9365_ 1", col1_3, col1_4, "co'l1_""5", c̝ol1_9), UNIQUE (lower("col1\\U000B9365_ 1") ASC, c̝ol1_9 ASC), INDEX ("co'l1_""5") STORING ("col1\\U000B9365_ 1", "c\\U000CA96Fol1_2", col1_4, "Col1 _
        6", co̿l1_8, c̝ol1_9) WHERE table1."Col1 _
        6" < '"':::STRING, INDEX (lower("col1\\U000B9365_ 1") ASC) WHERE (((((((((table1."Col1 _
        6" > '"':::STRING) OR (table1.col1_7 = 'X':::STRING)) OR (table1.col1_3 = 1:::INT8)) OR (table1.col1_0 <= '1970-01-01':::DATE)) AND (table1.co̿l1_8 <= (-1):::INT8)) OR (table1."col1\\U000B9365_ 1" < '':::STRING)) AND (table1.col1_4 = e'\x00':::STRING:::NAME)) OR (table1."col1_�10" != e'\U00002603':::STRING)) AND (table1.c̝ol1_9 >= 127:::INT8)) AND (table1."c\\U000CA96Fol1_2" != (-1):::INT8), UNIQUE ("Col1 _
        6" DESC) STORING (col1_0), UNIQUE ("Col1 _
        6" ASC, "c\\U000CA96Fol1_2" ASC) STORING (col1_0, "col1\\U000B9365_ 1", co̿l1_8), FAMILY (c̝ol1_9, "Col1 _
        6"), FAMILY (col1_3), FAMILY ("c\\U000CA96Fol1_2", co̿l1_8, "col1\\U000B9365_ 1", col1_4, "co'l1_""5", col1_0));
    rsg_test.go:722: CREATE TABLE table2 (col2_0 BYTES NOT NULL, "}col2_1" GEOMETRY NOT NULL, col2_2 GEOMETRY NOT NULL, col2_3 STRING NULL, "'col2 _4" INET NULL, "col2 _5" TIMETZ NOT NULL, col2_6 STRING NOT NULL AS (lower(CAST("}col2_1" AS STRING))) STORED, col2_7 STRING NOT NULL AS (lower(CAST("col2 _5" AS STRING))) VIRTUAL, "col2_	8" STRING NULL AS (lower(CAST("'col2 _4" AS STRING))) VIRTUAL, PRIMARY KEY ("col2 _5" DESC), INDEX ((CASE WHEN col2_0 IS NULL THEN e'\x02e~\x17V':::STRING ELSE e'\x07>68qE7':::STRING END), col2_6 DESC, col2_3), INVERTED INDEX (col2_2), INDEX (col2_3 DESC) STORING (col2_0, col2_6), UNIQUE ("col2 _5" DESC) STORING ("}col2_1", col2_2, col2_3), INVERTED INDEX ("col2 _5" DESC, "}col2_1"), UNIQUE (col2_0, "'col2 _4" ASC) STORING ("}col2_1", col2_3, col2_6) WHERE ((table2.col2_0 != '\x58':::BYTES) OR (table2.col2_3 != e'\x00':::STRING)) OR (table2.col2_7 >= e'\U00002603':::STRING), INVERTED INDEX (col2_2 ASC) WHERE (((((table2."col2 _5" >= '00:00:00+15:59':::TIMETZ) OR (table2.col2_6 = '"':::STRING)) AND (table2.col2_3 = e'\x00':::STRING)) AND (table2."col2_	8" >= '"':::STRING)) AND (table2.col2_7 >= e'\U00002603':::STRING)) AND (table2.col2_0 < '\xff':::BYTES), FAMILY (col2_2), FAMILY (col2_6), FAMILY ("col2 _5"), FAMILY (col2_3, "}col2_1", "'col2 _4"), FAMILY (col2_0), UNIQUE (col2_3));
    rsg_test.go:722: CREATE TABLE "ta̹""blE%v3" ("cOl)3_0" OID, c😾ol3_1 UUID NOT NULL, col3_2 DECIMAL[] NOT NULL, "col3'_3" INT2 NOT NULL, col3_4 TIMESTAMPTZ NULL, UNIQUE (col3_4 ASC), INDEX (col3_2 DESC), INDEX (col3_4 DESC), INVERTED INDEX (col3_2 ASC) WHERE ("ta̹""blE%v3".col3_4 < '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ) AND ("ta̹""blE%v3"."col3'_3" >= 1:::INT8), INDEX (col3_4 DESC, col3_2 ASC) STORING ("cOl)3_0", c😾ol3_1, "col3'_3"), FAMILY ("col3'_3"), FAMILY (col3_4), FAMILY (c😾ol3_1, "cOl)3_0", col3_2));
    rsg_test.go:722: CREATE TABLE table4 ("c'ol4_0" REFCURSOR NOT NULL, "co%pl4_1" STRING COLLATE da_DK NOT NULL, "co\\U0002A8F0l4_2" STRING NOT NULL, PRIMARY KEY ("co%pl4_1" ASC), INDEX ("co%pl4_1" DESC) STORING ("co\\U0002A8F0l4_2") WHERE table4."co\\U0002A8F0l4_2" <= e'\x00':::STRING, UNIQUE ("co\\U0002A8F0l4_2" DESC, "co%pl4_1" ASC) STORING ("c'ol4_0"), INDEX ("co\\U0002A8F0l4_2", lower("co\\U0002A8F0l4_2") ASC, "co%pl4_1" DESC), INVERTED INDEX ("co\\U0002A8F0l4_2" gin_trgm_ops ASC), INDEX ("co%pl4_1") STORING ("c'ol4_0", "co\\U0002A8F0l4_2") WHERE table4."co\\U0002A8F0l4_2" > e'\x00':::STRING, UNIQUE ("co%pl4_1" DESC) STORING ("co\\U0002A8F0l4_2"));
    rsg_test.go:722: ALTER TABLE table1 INJECT STATISTICS e'[{"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"co\'l1_\\"\\"5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 50000000, "num_range": 0, "upper_bound": "00100111101101000000011101100111"}, {"distinct_range": 21486.373797708336, "num_eq": 100000, "num_range": 40000, "upper_bound": "10111110001011111101101100010000"}, {"distinct_range": 0, "num_eq": 6395278812531229565, "num_range": 100000, "upper_bound": "11000000110001110001001011100011"}], "histo_col_type": "BIT(32)", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"Col1 _\\n6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 2917353429946228089, "num_range": 0, "upper_bound": "OU"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["c\u031Dol1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"c\\\\\\\\U000CA96Fol1_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["co\u033Fl1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col1_\\u000c10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 4309253652139751576, "num_range": 0, "upper_bound": ""}, {"distinct_range": 10000000, "num_eq": 9141051361773368220, "num_range": 10000000, "upper_bound": "\'"}, {"distinct_range": 1101881556815530800, "num_eq": 200000, "num_range": 6837251887642755379, "upper_bound": ".w:hof"}, {"distinct_range": 0, "num_eq": 6487837210681483270, "num_range": 2327692773508848042, "upper_bound": "X"}, {"distinct_range": 0, "num_eq": 100000000, "num_range": 4682832384367496932, "upper_bound": "kp!K~|\\u0008O"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col1\\\\\\\\U000B9365_ 1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}]':::JSONB;
    rsg_test.go:722: ALTER TABLE table2 INJECT STATISTICS e'[{"avg_size": 0, "columns": ["\\"}col2_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 5666981744376900, "num_range": 0, "upper_bound": "\\\\x42fd100000000000000000c201f5603220eeb2c201b85160c0e83842017431cac8b7b242027c81b0b2f140"}, {"distinct_range": 694032377.2924738, "num_eq": 200000000, "num_range": 1000000000, "upper_bound": "\\\\x42fdffffffffffffffff00c201f5603220eeb2c201b85160c0e83842017431cac8b7b242027c81b0b2f140"}], "histo_col_type": "BYTES", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"\'col2 _4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 700000, "num_range": 0, "upper_bound": "\\\\x0d8b10f6"}], "histo_col_type": "BYTES", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 9000000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 2093646313928398603, "num_range": 9000000000, "upper_bound": "\\u0004nsz"}, {"distinct_range": 167366.22657879366, "num_eq": 200000, "num_range": 200000, "upper_bound": "\\u001f\\u0001"}, {"distinct_range": 673966319809162800, "num_eq": 0, "num_range": 3466775806450503572, "upper_bound": "\'mDY"}, {"distinct_range": 2376146060867016700, "num_eq": 100000000, "num_range": 2376146060867016559, "upper_bound": "7/\\u000f[q1"}, {"distinct_range": 5000000000, "num_eq": 5400646993979689434, "num_range": 5000000000, "upper_bound": "N\\u0001m"}, {"distinct_range": 0, "num_eq": 90, "num_range": 0, "upper_bound": "X"}, {"distinct_range": 100, "num_eq": 8000, "num_range": 100, "upper_bound": "_HLCt"}, {"distinct_range": 0, "num_eq": 9000, "num_range": 7597807440346093437, "upper_bound": "y\\u0005\\rv_)<-"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col2 _5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 500, "num_range": 0, "upper_bound": "02:19:16.977275+12:15"}, {"distinct_range": 0, "num_eq": 8000000, "num_range": 100000000, "upper_bound": "09:43:04.476674+14:22"}, {"distinct_range": 2361738735435222500, "num_eq": 8048207387168615842, "num_range": 2361738735435222576, "upper_bound": "10:40:21.738228+11:43"}, {"distinct_range": 7033443982332438000, "num_eq": 7252505441742751082, "num_range": 7033443982332437322, "upper_bound": "17:33:37.835802+11:52"}, {"distinct_range": 778205955189133300, "num_eq": 4379489097248281706, "num_range": 4437326438962682095, "upper_bound": "13:24:02.618497-07:27"}, {"distinct_range": 6395816727989742000, "num_eq": 500000, "num_range": 6395816727989741908, "upper_bound": "19:00:02.653292-03:48"}, {"distinct_range": 3598125867152915000, "num_eq": 4621018065919864887, "num_range": 3598125867152914776, "upper_bound": "13:35:57.142947-11:08"}, {"distinct_range": 3277513996903198700, "num_eq": 700000000, "num_range": 3277513996903198956, "upper_bound": "23:01:27.327182-08:34"}, {"distinct_range": 0, "num_eq": 6624828147144696739, "num_range": 0, "upper_bound": "20:45:36.048549-13:03"}], "histo_col_type": "TIMETZ", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col2_\\t8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}]':::JSONB;
    rsg_test.go:722: ALTER TABLE "ta̹""blE%v3" INJECT STATISTICS e'[{"avg_size": 12, "columns": ["\\"col3\'_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 599128279145443169, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8549797367575342777}, {"avg_size": 10, "columns": ["col3_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8433145682482100796, "histo_col_type": "TIMESTAMPTZ", "name": "__auto__", "null_count": 7870938027699497444, "row_count": 8549797367575342777}, {"avg_size": 14, "columns": ["\\"cOl)3_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2113887338349801270, "histo_col_type": "", "name": "__auto__", "null_count": 3569008321625947713, "row_count": 8549797367575342777}, {"avg_size": 13, "columns": ["c\U0001F63Eol3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 483383060388583160, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8549797367575342777}, {"avg_size": 24, "columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7481984364235718100, "histo_col_type": "DECIMAL[]", "name": "__auto__", "null_count": 0, "row_count": 8549797367575342777}]':::JSONB;
    rsg_test.go:722: ALTER TABLE table4 INJECT STATISTICS e'[{"avg_size": 18, "columns": ["\\"c\'ol4_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4271548373, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000000}, {"avg_size": 9, "columns": ["\\"co%pl4_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4299659244, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000000}, {"avg_size": 29, "columns": ["\\"co\\\\\\\\U0002A8F0l4_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3188697453, "histo_buckets": [{"distinct_range": 0, "num_eq": 8351473780765734996, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 3237223750289643133, "num_range": 500, "upper_bound": " ?\\u0003D.)L<"}, {"distinct_range": 214907290.75337628, "num_eq": 0, "num_range": 5000000000, "upper_bound": "<ujB3"}, {"distinct_range": 30000000, "num_eq": 8789349280571330646, "num_range": 30000000, "upper_bound": "A+&C\\tV\\u00064"}, {"distinct_range": 2516169823298782000, "num_eq": 0, "num_range": 2516169823298782318, "upper_bound": "Dg=\\\\\\u0008\\n(_\x7f"}, {"distinct_range": 90000000, "num_eq": 2374478019572720695, "num_range": 90000000, "upper_bound": "fRIA\\u0010u3_"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 5000000000}]':::JSONB;
    rsg_test.go:722: ALTER TABLE table1 ADD FOREIGN KEY ("col1\\U000B9365_ 1") REFERENCES table2 (col2_3) ON UPDATE CASCADE;
    rsg_test.go:722: CREATE TYPE rand_typ_0 AS ENUM ('gpdqu');
    rsg_test.go:722: CREATE TYPE rand_typ_1 AS ENUM ('gy', 'qf', 'gfx', 'am');
    rsg_test.go:722: CREATE TYPE rand_typ_2 AS ENUM ('mtmr', 'aajf', 'gh', 'yzxpei');
    rsg_test.go:722: SET sql_safe_updates = false;;
    rsg_test.go:722: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:722: ALTER TABLE seed SPLIT AT VALUES (917209283323428038);
    rsg_test.go:722: ALTER TABLE seed SPLIT AT VALUES (6289241433342398576);
    rsg_test.go:722: ALTER TABLE seed SPLIT AT VALUES (3788321235458868234);
    rsg_test.go:722: ALTER TABLE seed SPLIT AT VALUES (4819656368278249256);
    rsg_test.go:722: ALTER TABLE seed SPLIT AT VALUES (6220318643998670157);
    rsg_test.go:722: ALTER TABLE seed SCATTER;
    rsg_test.go:724: 
    rsg_test.go:725: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith3059716433
--- FAIL: TestRandomSyntaxSQLSmith (381.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119188 sql/tests: TestRandomSyntaxSQLSmith failed [expected exactly one ordering column for RANGE mode with offset] [A-sql-optimizer C-bug C-test-failure O-robot P-3 T-sql-queries branch-release-23.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36197

--- a/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_query_behavior
+++ b/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_query_behavior
@@ -3647,7 +3647,7 @@ project
  ├── immutable
  ├── stats: [rows=1.031997]
  ├── cost: 122.172083
- ├── fd: ()-->(2,5,17,20)
+ ├── fd: ()-->(2,5,17,20), (5)==(20), (20)==(5)
  ├── distribution: ap-southeast-2
  ├── prune: (2-5,10,17-20)
  └── inner-join (lookup xyz [as=x])

--- a/pkg/sql/opt/memo/testdata/logprops/join
+++ b/pkg/sql/opt/memo/testdata/logprops/join
@@ -148,7 +148,7 @@ project
       │              ├── columns: v:8(int!null) n:13(int!null)
       │              ├── outer: (1)
       │              ├── multiplicity: left-rows(zero-or-one), right-rows(zero-or-more)
-      │              ├── fd: ()-->(8,13)
+      │              ├── fd: ()-->(8,13), (8)==(13), (13)==(8)
       │              ├── interesting orderings: (+13)
       │              ├── scan uv
       │              │    ├── columns: v:8(int!null)
@@ -2913,3 +2913,111 @@ left-join (hash)
       └── eq [type=bool, outer=(2,7), constraints=(/2: (/NULL - ]; /7: (/NULL - ]), fd=(2)==(7), (7)==(2)]
            ├── variable: r_b:2 [type=int]
            └── variable: b:7 [type=int]
+
+# Regression test for #119441 - marking a set of columns constant should not
+# remove equalities among them.
+exec-ddl
+CREATE TABLE table1_119441 (col1_5 REGTYPE NULL, col1_8 BYTES, col1_11 STRING, col1_12 STRING, INDEX (col1_11));
+----
+
+exec-ddl
+CREATE TABLE table2_119441 (col2_1 INT4, col2_2 STRING, col2_3 BYTES);
+----
+
+exec-ddl
+CREATE TABLE table3_119441 (col3_5 VARCHAR NOT NULL, col3_11 CHAR, col3_16 INT8);
+----
+
+opt
+SELECT (
+  SELECT baz.col3_5 FROM table1_119441
+  JOIN table2_119441 ON (table1_119441.col1_8) = (table2_119441.col2_3) AND (table1_119441.col1_12) = (table2_119441.col2_2) AND (table1_119441.col1_11) = (table2_119441.col2_2)
+  JOIN (SELECT baz.col3_11, table3_119441.col3_16 FROM table3_119441 ORDER BY table3_119441.col3_16 LIMIT 6) AS t(foo, bar)
+  ON table1_119441.col1_11 = foo AND table2_119441.col2_1 = bar AND table1_119441.col1_12 = foo
+  LIMIT 1
+)
+FROM table3_119441 AS baz;
+----
+project
+ ├── columns: col3_5:28(varchar)
+ ├── prune: (28)
+ ├── reject-nulls: (28)
+ ├── left-join-apply
+ │    ├── columns: baz.col3_5:1(varchar!null) baz.col3_11:2(char) col3_5:27(varchar)
+ │    ├── prune: (27)
+ │    ├── reject-nulls: (27)
+ │    ├── scan table3_119441 [as=baz]
+ │    │    ├── columns: baz.col3_5:1(varchar!null) baz.col3_11:2(char)
+ │    │    ├── prune: (1,2)
+ │    │    └── unfiltered-cols: (1-6)
+ │    ├── project
+ │    │    ├── columns: col3_5:27(varchar)
+ │    │    ├── outer: (1,2)
+ │    │    ├── cardinality: [0 - 1]
+ │    │    ├── key: ()
+ │    │    ├── fd: ()-->(27)
+ │    │    ├── prune: (27)
+ │    │    ├── limit
+ │    │    │    ├── columns: col1_8:8(bytes!null) col1_11:9(string!null) col1_12:10(string!null) col2_1:14(int4!null) col2_2:15(string!null) col2_3:16(bytes!null) table3_119441.col3_16:22(int!null)
+ │    │    │    ├── outer: (2)
+ │    │    │    ├── cardinality: [0 - 1]
+ │    │    │    ├── key: ()
+ │    │    │    ├── fd: ()-->(8-10,14-16,22), (10)==(9,15), (8)==(16), (16)==(8), (15)==(9,10), (14)==(22), (22)==(14), (9)==(10,15)
+ │    │    │    ├── interesting orderings: (+(9|10)) (+22)
+ │    │    │    ├── inner-join (hash)
+ │    │    │    │    ├── columns: col1_8:8(bytes!null) col1_11:9(string!null) col1_12:10(string!null) col2_1:14(int4!null) col2_2:15(string!null) col2_3:16(bytes!null) table3_119441.col3_16:22(int!null)
+ │    │    │    │    ├── outer: (2)
+ │    │    │    │    ├── fd: ()-->(9,10,15), (9)==(10,15), (10)==(9,15), (8)==(16), (16)==(8), (15)==(9,10), (14)==(22), (22)==(14)
+ │    │    │    │    ├── limit hint: 1.00
+ │    │    │    │    ├── interesting orderings: (+(9|10)) (+22)
+ │    │    │    │    ├── top-k
+ │    │    │    │    │    ├── columns: table3_119441.col3_16:22(int)
+ │    │    │    │    │    ├── internal-ordering: +22
+ │    │    │    │    │    ├── k: 6
+ │    │    │    │    │    ├── cardinality: [0 - 6]
+ │    │    │    │    │    ├── interesting orderings: (+22)
+ │    │    │    │    │    └── scan table3_119441
+ │    │    │    │    │         ├── columns: table3_119441.col3_16:22(int)
+ │    │    │    │    │         └── prune: (22)
+ │    │    │    │    ├── inner-join (hash)
+ │    │    │    │    │    ├── columns: col1_8:8(bytes!null) col1_11:9(string!null) col1_12:10(string!null) col2_1:14(int4) col2_2:15(string!null) col2_3:16(bytes!null)
+ │    │    │    │    │    ├── fd: (9)==(10,15), (10)==(9,15), (8)==(16), (16)==(8), (15)==(9,10)
+ │    │    │    │    │    ├── prune: (14)
+ │    │    │    │    │    ├── interesting orderings: (+(9|10))
+ │    │    │    │    │    ├── scan table2_119441
+ │    │    │    │    │    │    ├── columns: col2_1:14(int4) col2_2:15(string) col2_3:16(bytes)
+ │    │    │    │    │    │    ├── prune: (14-16)
+ │    │    │    │    │    │    └── unfiltered-cols: (14-19)
+ │    │    │    │    │    ├── select
+ │    │    │    │    │    │    ├── columns: col1_8:8(bytes) col1_11:9(string!null) col1_12:10(string!null)
+ │    │    │    │    │    │    ├── fd: (9)==(10), (10)==(9)
+ │    │    │    │    │    │    ├── prune: (8)
+ │    │    │    │    │    │    ├── interesting orderings: (+(9|10))
+ │    │    │    │    │    │    ├── scan table1_119441
+ │    │    │    │    │    │    │    ├── columns: col1_8:8(bytes) col1_11:9(string) col1_12:10(string)
+ │    │    │    │    │    │    │    ├── prune: (8-10)
+ │    │    │    │    │    │    │    └── interesting orderings: (+9)
+ │    │    │    │    │    │    └── filters
+ │    │    │    │    │    │         └── eq [type=bool, outer=(9,10), constraints=(/9: (/NULL - ]; /10: (/NULL - ]), fd=(9)==(10), (10)==(9)]
+ │    │    │    │    │    │              ├── variable: col1_11:9 [type=string]
+ │    │    │    │    │    │              └── variable: col1_12:10 [type=string]
+ │    │    │    │    │    └── filters
+ │    │    │    │    │         ├── eq [type=bool, outer=(8,16), constraints=(/8: (/NULL - ]; /16: (/NULL - ]), fd=(8)==(16), (16)==(8)]
+ │    │    │    │    │         │    ├── variable: col1_8:8 [type=bytes]
+ │    │    │    │    │         │    └── variable: col2_3:16 [type=bytes]
+ │    │    │    │    │         └── eq [type=bool, outer=(9,15), constraints=(/9: (/NULL - ]; /15: (/NULL - ]), fd=(9)==(15), (15)==(9)]
+ │    │    │    │    │              ├── variable: col1_11:9 [type=string]
+ │    │    │    │    │              └── variable: col2_2:15 [type=string]
+ │    │    │    │    └── filters
+ │    │    │    │         ├── eq [type=bool, outer=(14,22), constraints=(/14: (/NULL - ]; /22: (/NULL - ]), fd=(14)==(22), (22)==(14)]
+ │    │    │    │         │    ├── variable: col2_1:14 [type=int4]
+ │    │    │    │         │    └── variable: table3_119441.col3_16:22 [type=int]
+ │    │    │    │         └── eq [type=bool, outer=(2,9), constraints=(/2: (/NULL - ]; /9: (/NULL - ]), fd=(2)==(9), (9)==(2)]
+ │    │    │    │              ├── variable: col1_11:9 [type=string]
+ │    │    │    │              └── variable: baz.col3_11:2 [type=char]
+ │    │    │    └── const: 1 [type=int]
+ │    │    └── projections
+ │    │         └── variable: baz.col3_5:1 [as=col3_5:27, type=varchar, outer=(1)]
+ │    └── filters (true)
+ └── projections
+      └── variable: col3_5:27 [as=col3_5:28, type=varchar, outer=(27)]

--- a/pkg/sql/opt/props/func_dep.go
+++ b/pkg/sql/opt/props/func_dep.go
@@ -1088,7 +1088,7 @@ func (f *FuncDepSet) ProjectCols(cols opt.ColSet) {
 		// constant columns from dependants for nicer presentation.
 		if !fd.to.SubsetOf(cols) {
 			fd.to = fd.to.Intersection(cols)
-			if !fd.isConstant() {
+			if !fd.isConstant() && !fd.equiv {
 				fd.to.DifferenceWith(constCols)
 			}
 			if !fd.removeToCols(fd.from) {

